core/src/test/java/org/apache/accumulo/core/data/RowRangeTest.java

```diff
@@ -0,0 +1,569 @@
+      RowRange range4 = RowRange.atLeast(new Text("row8"));
+      RowRange range5 = RowRange.lessThan("r2");
+      RowRange range6 = RowRange.lessThan(new Text("r2"));
+      RowRange range7 = RowRange.atMost("r3");
+      RowRange range8 = RowRange.atMost(new Text("r3"));
+
+      // Test that all ranges created using different entry point methods are equal
+      assertTrue(range1.equals(range2));
+      assertTrue(range3.equals(range4));
+      assertTrue(range5.equals(range6));
+      assertTrue(range7.equals(range8));
+
+      // Test that ranges with different properties are not equal
+      assertFalse(range1.equals(range3));
+      assertFalse(range1.equals(range5));
+      assertFalse(range1.equals(range7));
+      assertFalse(range3.equals(range5));
+      assertFalse(range3.equals(range7));
+      assertFalse(range5.equals(range7));
+    }
+  }
+
+  @Nested
+  class CompareToTests {
+
+    @Test
+    void testCompareWithSameRange() {
+      RowRange range1 = RowRange.open("r1", "r3");
+      RowRange range2 = RowRange.open("r1", "r3");
+      assertEquals(0, range1.compareTo(range2));
+    }
+
+    @Test
+    void testCompareWithDifferentStartRow() {
+      RowRange range1 = RowRange.open("r1", "r3");
+      RowRange range2 = RowRange.open("r2", "r3");
+      assertTrue(range1.compareTo(range2) < 0);
+    }
+
+    @Test
+    void testCompareWithDifferentEndRow() {
+      RowRange range1 = RowRange.open("r1", "r3");
+      RowRange range2 = RowRange.open("r1", "r4");
+      assertTrue(range1.compareTo(range2) < 0);
+    }
+
+    @Test
+    void testCompareWithDifferentStartRowInclusiveness() {
+      RowRange range1 = RowRange.open("r1", "r3");
+      RowRange range2 = RowRange.closedOpen("r1", "r3");
+      assertTrue(range1.compareTo(range2) > 0);
+    }
+
+    @Test
+    void testCompareWithDifferentEndRowInclusiveness() {
+      RowRange range1 = RowRange.open("r1", "r3");
+      RowRange range2 = RowRange.openClosed("r1", "r3");
+      assertTrue(range1.compareTo(range2) < 0);
+    }
+
+    @Test
+    void testCompareWithInfiniteStartRow() {
+      RowRange range1 = RowRange.atLeast("r1");
+      RowRange range2 = RowRange.all();
+      assertTrue(range1.compareTo(range2) < 0);
+    }
+
+    @Test
+    void testCompareWithInfiniteEndRow() {
+      RowRange range1 = RowRange.all();
+      RowRange range2 = RowRange.atLeast("r1");
+      assertTrue(range1.compareTo(range2) > 0);
+    }
+  }
+
+  @Nested
+  class ContainsTests {
+
+    @Test
+    void testContainsWithAllRange() {
+      RowRange range = RowRange.all();
+      assertTrue(range.contains(new Text("r1")));
+      assertTrue(range.contains(new Text("r2")));
+      assertTrue(range.contains(new Text("row3")));
+    }
+
+    @Test
+    void testContainsWithOpenRange() {
+      RowRange range = RowRange.open("r1", "r3");
+      assertFalse(range.contains(new Text("r1")));
+      assertTrue(range.contains(new Text("r2")));
+      assertFalse(range.contains(new Text("r3")));
```

Comment:
Added some test data w/ same prefix and with binary zero that is next possible row.

```suggestion
      assertFalse(range.contains(new Text("r1")));
      assertTrue(range.contains(new Text(new byte[]{'r','1',0})));
      assertTrue(range.contains(new Text("r11")));
      assertTrue(range.contains(new Text("r2")));
      assertFalse(range.contains(new Text(new byte[]{'r','3',0})));
      assertFalse(range.contains(new Text("r3")));
```